pkg/test/spacebindingrequest/spacebindingrequest.go

```diff
@@ -0,0 +1,66 @@
+package spacebindingrequest
+
+import (
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/gofrs/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type Option func(spaceRequest *toolchainv1alpha1.SpaceBindingRequest)
+
+func NewSpaceBindingRequest(name, namespace string, options ...Option) *toolchainv1alpha1.SpaceBindingRequest {
+	spaceBindingRequest := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			UID:       types.UID(uuid.Must(uuid.NewV4()).String()),
+		},
+	}
+	for _, apply := range options {
+		apply(spaceBindingRequest)
+	}
+	return spaceBindingRequest
+}
+
+func WithMUR(mur string) Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Spec.MasterUserRecord = mur
+	}
+}
+
+func WithSpaceRole(spaceRole string) Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Spec.SpaceRole = spaceRole
+	}
+}
+
+func WithLabel(key, value string) Option {
+	return func(space *toolchainv1alpha1.SpaceBindingRequest) {
+		if space.ObjectMeta.Labels == nil {
+			space.ObjectMeta.Labels = map[string]string{}
+		}
+		space.ObjectMeta.Labels[key] = value
+	}
+}
+
+func WithDeletionTimestamp() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		now := metav1.NewTime(time.Now())
+		spaceBindingRequest.DeletionTimestamp = &now
+	}
+}
+
+func WithFinalizer() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Finalizers = append(spaceBindingRequest.Finalizers, toolchainv1alpha1.FinalizerName)
+	}
+}
```

Comment:
Should this check if the given finalizer is already set?

Reply:
may [controller-runtime's AddFinalizer](https://github.com/kubernetes-sigs/controller-runtime/blob/984a121dbb8f6273eef55ebb4016e3b3deef1919/pkg/controller/controllerutil/controllerutil.go#L440) be helpful here?

Reply:
makes sense. Let's use the AddFinalizer util , thanks for the suggestions! 

Please see https://github.com/codeready-toolchain/toolchain-common/pull/351/commits/b0db03ef3fa2c54d02152e3554f831606266756c